docs/src/dev-guide/components-core/regex-utils.md

```diff
@@ -62,6 +62,19 @@ For a detailed description on the options order and usage, see the
   * Turn `.*` into `*`
   * Turn `.+` into `?*`
   * E.g. `abc.*def.ghi.+` will get translated to `abc*def?ghi?*`
+* Metacharacter escape sequences
+  * Append the escaped metacharacters literally to the wildcard output string.
+    * The list of characters that require escaping is `[\/^$.|?*+(){}`.
+    * The list of characters that are optional to escape is `],<>-_=!`. Escaping them will not have
+      any effect (the escape backslash is ignored). This list may be expanded in the future.
```

Comment:
I see what you're saying here, but this is standard behaviour so I think we can reword it a bit so it doesn't sound like we're doing something different.
Generally speaking, escaping any character that doesn't result in a special sequence has no effect. Different libraries go to different lengths to recognize when the characters don't turn into a special sequence.

Lets switch to something like:
"Superfluous escape characters are ignored for the following characters: `],<>-_=!`."

Mentioning we may expand this list in the future is optional. You can keep it if you like.